x/evm/keeper/integration_test.go

```diff
@@ -1,246 +1,252 @@
-					},
-					Entry("legacy tx", func() txParams {
-						return txParams{0, big.NewInt(baseFee), nil, nil, nil}
-					}),
-					Entry("dynamic tx", func() txParams {
-						return txParams{0, nil, big.NewInt(baseFee), big.NewInt(0), &ethtypes.AccessList{}}
-					}),
-				)
-			})
-
-			//nolint:all
-			Context("during DeliverTx", func() {
-				DescribeTable("should accept transactions with gas Limit > 0",
-					func(malleate getprices) {
-						p := malleate()
-						to := utiltx.GenerateAddress()
-						msgEthereumTx := buildEthTx(privKey, &to, p.gasLimit, p.gasPrice, p.gasFeeCap, p.gasTipCap, p.accesses)
-						res, err := testutil.DeliverEthTx(s.app, privKey, msgEthereumTx)
-						Expect(err).To(BeNil())
-						Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
-					},
-					Entry("legacy tx", func() txParams {
-						return txParams{100000, big.NewInt(baseFee), nil, nil, nil}
-					}),
-					Entry("dynamic tx", func() txParams {
-						return txParams{100000, nil, big.NewInt(baseFee), big.NewInt(0), &ethtypes.AccessList{}}
-					}),
-				)
-				DescribeTable("should not accept transactions with gas Limit > 0",
-					func(malleate getprices) {
-						p := malleate()
-						to := utiltx.GenerateAddress()
-						msgEthereumTx := buildEthTx(privKey, &to, p.gasLimit, p.gasPrice, p.gasFeeCap, p.gasTipCap, p.accesses)
-						res, err := testutil.DeliverEthTx(s.app, privKey, msgEthereumTx)
-						Expect(err).ToNot(BeNil(), "transaction should have failed", res.GetLog())
-					},
-					Entry("legacy tx", func() txParams {
-						return txParams{0, big.NewInt(baseFee), nil, nil, nil}
-					}),
-					Entry("dynamic tx", func() txParams {
-						return txParams{0, nil, big.NewInt(baseFee), big.NewInt(0), &ethtypes.AccessList{}}
-					}),
-				)
-			})
+		It("performs an AccessListTx", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			receiver := s.keyring.GetKey(1)
+			accessList := &ethtypes.AccessList{{Address: receiver.Addr, StorageKeys: []common.Hash{{0}}}}
+			// GasFeeCap and GasTipCap are populated by default by the factory
+			txArgs := evmtypes.EvmTxArgs{
+				To:       &receiver.Addr,
+				Amount:   big.NewInt(1000),
+				Accesses: accessList,
+			}
+
+			res, err := s.factory.ExecuteEthTx(senderPriv, txArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
+		})
+
+		It("performs a LegacyTx", func() {
+			senderPriv := s.keyring.GetPrivKey(0)
+			receiver := s.keyring.GetKey(1)
+			// GasFeeCap and GasTipCap are populated by default by the factory
+			txArgs := evmtypes.EvmTxArgs{
+				To:       &receiver.Addr,
+				Amount:   big.NewInt(1000),
+				GasPrice: big.NewInt(1e9),
+			}
+
+			res, err := s.factory.ExecuteEthTx(senderPriv, txArgs)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
```

Comment:
ditto

Reply:
Here I am just checking for the transaction not to be rejected, you think we should still check all the details of the tx that are already checked before?